instagram-app/src/components/PostContainer/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import Likes from './Likes/Likes';
+import PostedComments from './PostedComments/PostedComments';
+import AddComment from './AddComment/AddComment';
+import PropTypes from 'prop-types';
+import { CommentSectionDiv } from '../CommentSection/CommentSectionStyles'
+
+class CommentSection extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            comments: this.props.comments,
+            currentComment: {username: this.props.comments.username, text: this.props.comments.text},
+            likesCounter: this.props.likes,
+            click: true,
+            showForm: false
+        }
+    }
+
+    handleInput = e => {
+        this.setState({
+            currentComment: {...this.state.currentComment, [e.target.name]: e.target.value }
+        });
+    }
+
+    toggleClick = () => {
+        this.setState({ click: !this.state.click })
+        if (this.state.click === true) {
+            this.setState({ likesCounter: this.state.likesCounter + 1});
+        } else {
+            this.setState({ likesCounter: this.state.likesCounter - 1});
+        }
+    }
+    
+    addComment = e => {
+        e.preventDefault();
+        const newComment = this.state.currentComment
+        const comments = [...this.state.comments, newComment]
+        this.setState({
+            comments: comments,
+            currentComment: { username: '', text: ''}
+        });
+    }
+
+    toggleCommentForm = () => {
+        this.setState({ showForm: !this.state.showForm})
+    }
+
+    render() {
+        return (
+            <CommentSectionDiv>
+                <Likes 
+                    likes={this.state.likesCounter}  
+                    toggleClick={this.toggleClick}
+                    toggleCommentForm={this.toggleCommentForm} />
+                    
+                {this.state.comments.map(comment => {
+                    return <PostedComments 
+                    username={comment.username} 
+                    text={comment.text} />
+                })}
+                <AddComment 
+                    handleInput={this.handleInput} 
+                    addComment={this.addComment} 
+                    text={this.state.currentComment.text} 
+                    username={this.state.currentComment.username} 
+                    toggleCommentForm={this.toggleCommentForm}
```

Comment:
At first I thought you didn't have the addComment implemented, but then I saw your toggle! Awesome job adding comments to this, and the user's ability to add their own username and comment. I would recommend not toggling the form, just so the UI of being able to add comments is clear, but awesome job getting it in there.

Also, since we had talked about the HOC and emphasizing other functions, I love your creative solution to inputting the username. Really well done, Leslee.